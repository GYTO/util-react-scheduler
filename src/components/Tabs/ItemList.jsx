// @flow
import * as React from 'react';
import { PRODUCTS } from '../../api/producs';
import type { Products } from '../types/Products';

type Props = {
    itemClick: (index: number) => void,
}

type State = {
    products: Products[]
}

const boxStyle = {
    border: '1px solid gray',
    width: '200px',
    display: 'flex',
};

const boxes = {
    display: 'flex',
    flexDirection: 'column',
};

class ItemList extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {
            products: PRODUCTS,
        };
    }

    render() {
        const { products } = this.state;
        const { itemClick } = this.props;
        return (
            <div style={boxes}>
                {products.map((item, index) => {
                    return <button key={index} style={boxStyle} onClick={() => itemClick(item.id)}>
                        <img src={item.img} alt={item.name} />
                        <h4>{item.name}</h4>
                    </button>;
                })}
            </div>
        );
    }
}

export default ItemList;
