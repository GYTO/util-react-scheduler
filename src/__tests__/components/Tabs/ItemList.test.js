import * as React from 'react';
import {shallow, mount} from 'enzyme';
import ItemList from '../../../components/Tabs/ItemList';

describe('ItemList', () => {
    let props;
    let shallowRender;
    let mountedRender;

    const getMounted = () => {
        if (!mountedRender) mountedRender = mount(<ItemList {...props} />);
        return mountedRender;
    };

    const getShallow = () => {
        if (!shallowRender) shallowRender = shallow(<ItemList {...props} />);
        return shallowRender;
    };

    beforeEach(() => {
        jest.clearAllMocks();

        props = {
            itemClick: jest.fn(),
        };

        shallowRender = undefined;
        mountedRender = undefined;
    });

    it('matches snapshot', () => {
        expect(getShallow()).toMatchSnapshot();
    });

    it('renders onClick and return a item id', () => {
        getShallow().find('button').first().prop('onClick')();
        expect(typeof getMounted().find('button').first().prop('onClick')).toBe('function');
        expect(props.itemClick).toHaveBeenLastCalledWith(1);
    });
});
