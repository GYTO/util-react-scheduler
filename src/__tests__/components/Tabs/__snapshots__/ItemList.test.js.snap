// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemList
    itemClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
          style={
            Object {
              "border": "1px solid gray",
              "display": "flex",
              "width": "200px",
            }
          }
        >
          <img
            alt="Random Name"
            src="https://source.unsplash.com/random/50x50"
          />
          <h4>
            Random Name
          </h4>
        </button>,
        <button
          onClick={[Function]}
          style={
            Object {
              "border": "1px solid gray",
              "display": "flex",
              "width": "200px",
            }
          }
        >
          <img
            alt="Random Name"
            src="https://source.unsplash.com/random/50x50"
          />
          <h4>
            Random Name
          </h4>
        </button>,
        <button
          onClick={[Function]}
          style={
            Object {
              "border": "1px solid gray",
              "display": "flex",
              "width": "200px",
            }
          }
        >
          <img
            alt="Random Name"
            src="https://source.unsplash.com/random/50x50"
          />
          <h4>
            Random Name
          </h4>
        </button>,
        <button
          onClick={[Function]}
          style={
            Object {
              "border": "1px solid gray",
              "display": "flex",
              "width": "200px",
            }
          }
        >
          <img
            alt="Random Name"
            src="https://source.unsplash.com/random/50x50"
          />
          <h4>
            Random Name
          </h4>
        </button>,
      ],
      "style": Object {
        "display": "flex",
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Random Name"
              src="https://source.unsplash.com/random/50x50"
            />,
            <h4>
              Random Name
            </h4>,
          ],
          "onClick": [Function],
          "style": Object {
            "border": "1px solid gray",
            "display": "flex",
            "width": "200px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Random Name",
              "src": "https://source.unsplash.com/random/50x50",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Random Name",
            },
            "ref": null,
            "rendered": "Random Name",
            "type": "h4",
          },
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Random Name"
              src="https://source.unsplash.com/random/50x50"
            />,
            <h4>
              Random Name
            </h4>,
          ],
          "onClick": [Function],
          "style": Object {
            "border": "1px solid gray",
            "display": "flex",
            "width": "200px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Random Name",
              "src": "https://source.unsplash.com/random/50x50",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Random Name",
            },
            "ref": null,
            "rendered": "Random Name",
            "type": "h4",
          },
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Random Name"
              src="https://source.unsplash.com/random/50x50"
            />,
            <h4>
              Random Name
            </h4>,
          ],
          "onClick": [Function],
          "style": Object {
            "border": "1px solid gray",
            "display": "flex",
            "width": "200px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Random Name",
              "src": "https://source.unsplash.com/random/50x50",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Random Name",
            },
            "ref": null,
            "rendered": "Random Name",
            "type": "h4",
          },
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Random Name"
              src="https://source.unsplash.com/random/50x50"
            />,
            <h4>
              Random Name
            </h4>,
          ],
          "onClick": [Function],
          "style": Object {
            "border": "1px solid gray",
            "display": "flex",
            "width": "200px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Random Name",
              "src": "https://source.unsplash.com/random/50x50",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Random Name",
            },
            "ref": null,
            "rendered": "Random Name",
            "type": "h4",
          },
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
            style={
              Object {
                "border": "1px solid gray",
                "display": "flex",
                "width": "200px",
              }
            }
          >
            <img
              alt="Random Name"
              src="https://source.unsplash.com/random/50x50"
            />
            <h4>
              Random Name
            </h4>
          </button>,
          <button
            onClick={[Function]}
            style={
              Object {
                "border": "1px solid gray",
                "display": "flex",
                "width": "200px",
              }
            }
          >
            <img
              alt="Random Name"
              src="https://source.unsplash.com/random/50x50"
            />
            <h4>
              Random Name
            </h4>
          </button>,
          <button
            onClick={[Function]}
            style={
              Object {
                "border": "1px solid gray",
                "display": "flex",
                "width": "200px",
              }
            }
          >
            <img
              alt="Random Name"
              src="https://source.unsplash.com/random/50x50"
            />
            <h4>
              Random Name
            </h4>
          </button>,
          <button
            onClick={[Function]}
            style={
              Object {
                "border": "1px solid gray",
                "display": "flex",
                "width": "200px",
              }
            }
          >
            <img
              alt="Random Name"
              src="https://source.unsplash.com/random/50x50"
            />
            <h4>
              Random Name
            </h4>
          </button>,
        ],
        "style": Object {
          "display": "flex",
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Random Name"
                src="https://source.unsplash.com/random/50x50"
              />,
              <h4>
                Random Name
              </h4>,
            ],
            "onClick": [Function],
            "style": Object {
              "border": "1px solid gray",
              "display": "flex",
              "width": "200px",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Random Name",
                "src": "https://source.unsplash.com/random/50x50",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Random Name",
              },
              "ref": null,
              "rendered": "Random Name",
              "type": "h4",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Random Name"
                src="https://source.unsplash.com/random/50x50"
              />,
              <h4>
                Random Name
              </h4>,
            ],
            "onClick": [Function],
            "style": Object {
              "border": "1px solid gray",
              "display": "flex",
              "width": "200px",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Random Name",
                "src": "https://source.unsplash.com/random/50x50",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Random Name",
              },
              "ref": null,
              "rendered": "Random Name",
              "type": "h4",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Random Name"
                src="https://source.unsplash.com/random/50x50"
              />,
              <h4>
                Random Name
              </h4>,
            ],
            "onClick": [Function],
            "style": Object {
              "border": "1px solid gray",
              "display": "flex",
              "width": "200px",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Random Name",
                "src": "https://source.unsplash.com/random/50x50",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Random Name",
              },
              "ref": null,
              "rendered": "Random Name",
              "type": "h4",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Random Name"
                src="https://source.unsplash.com/random/50x50"
              />,
              <h4>
                Random Name
              </h4>,
            ],
            "onClick": [Function],
            "style": Object {
              "border": "1px solid gray",
              "display": "flex",
              "width": "200px",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Random Name",
                "src": "https://source.unsplash.com/random/50x50",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Random Name",
              },
              "ref": null,
              "rendered": "Random Name",
              "type": "h4",
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
